feel/feeldiscr/operatorinterpolation.hpp

```diff
@@ -1074,30 +1258,26 @@ OperatorInterpolation<DomainSpaceType, ImageSpaceType,IteratorRange,InterpType>:
 
                                 for ( uint16_type jloc = 0; jloc < domain_basis_type::nLocalDof; ++jloc )
                                 {
-                                    uint16_type compDomain = (domain_basis_type::is_product)? comp : 0;
+                                    uint16_type nCompDomain = (domain_basis_type::is_product)?domain_basis_type::nComponents:1;
+                                    for(int cdomain = 0;cdomain < nCompDomain;++cdomain )
+                                    {
```

Comment:
I tested the following code 
```cpp
#include <feel/feel.hpp>                                                                                                                       
#include <feel/feeldiscr/ned1h.hpp>                                                                                                            
#include <feel/feeldiscr/operatorinterpolation.hpp>                                                                                            
                                                                                                                                               
/** use Feel namespace */                                                                                                                      
using namespace Feel;                                                                                                                          
                                                                                                                                               
                                                                                                                                               
int main(int argc, char** argv )                                                                                                               
{                                                                                                                                              
    Environment env( argc, argv );                                                                                                             
    auto mesh = loadMesh( _mesh=new Mesh<Simplex<2>> );                                                                                        
    auto Xh = Pchv<1>( mesh );                                                                                                                 
    auto Yh = Pchv<1>( mesh );                                                                                                                 
    auto Ih = I( _domainSpace = Xh, _imageSpace = Yh );                                                                                        
    Ih.matPtr()->printMatlab( "Ih.m");                                                                                                         
}  
````
it should give the identity matrix, which it gives.
However I had to fix two bugs, the local interpolant had the wrong size and the local degree of freedom in the image space from the domain space was also wrong. both were missing dealing with the number of components. 
Now I agree that in the case of Lagrange FE (domain and image) we can optimize that depending on the expression.
For example for `id()` we should skip the non-diagonal terms automatically, no coupling between the components. 
But be careful other operators could couple components (e.g. curl)